### PR TITLE
test(tui): expand dialog and screen validation tests (#438)

### LINKED ISSUE #438: test(tui): Expand dialog and screen validation tests
## Description

Expand TUI test coverage for dialog validation logic and screen-level behavior that isn't yet covered by the existing 459 TuiUnit tests.

## Context

This issue was originally scoped around "20% TUI coverage (5/25 files)" and targeted extracting `AuthMethodFormBuilder`, `SqlQueryPresenter`, and `TableKeyboardHandler` as testable units. The testing infrastructure review (2026-03-24) found:

- **459 TuiUnit tests** across **30 test files** now exist (up from 5 files originally)
- The specific extraction targets (AuthMethodFormBuilder, SqlQueryPresenter, TableKeyboardHandler) were never created — the refactoring took a different approach
- TUI snapshot E2E tests cover startup and SQL Query screen (12 tests, 4 snapshots)

The original premise is outdated. This re-scoped issue targets remaining gaps.

### Dialog validation
- Profile creation dialog validation (auth method field requirements, URL format)
- Connection dialog state management
- Filter dialog state and validation

### Screen-level behavior
- Plugin Registration screen logic (not just SQL Query)
- Import Jobs screen behavior
- Solutions screen behavior

## Acceptance Criteria

- [ ] Dialog validation logic has unit tests for field requirements and error states
- [ ] At least 2 additional TUI screens have extracted logic tests
- [ ] All tests use `[Trait("Category", "TuiUnit")]`
- [ ] All tests pass with `dotnet test --filter "Category=TuiUnit"`

## Notes

- Re-scoped from v1.0 to v1.1 — existing 459 TuiUnit tests + snapshot tests provide adequate v1 regression coverage
- Original issue assumed specific type extractions that didn't happen — scope now targets whatever testable logic exists

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary
- Add ProfileCreationDialogState tests (state record, whitespace validation, trim behavior, method count)
- Add PluginTraceFilter tests (record construction, mode/date/correlation filters, equality semantics)
- Add SolutionsScreenState tests (loading/filtered/error/component states for Solutions screen)
- Add PluginTreeNode tests (icon mapping for all 10 node types, disabled/managed suffixes, tree structure)

Closes #438

## Test Plan
- [x] All 52 new tests pass across net8.0, net9.0, net10.0
- [x] Full TuiUnit suite passes (578 tests, 0 failures)
- [x] Full solution test suite passes (0 failures)

## Verification
- [x] /gates passed (build + tests green)
- [x] No production code changed — test-only PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/PPDS.Cli.Tests/Tui/Dialogs/PluginTraceFilterTests.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        Assert.Null(filter.TypeName);
+        Assert.Null(filter.MessageName);
+        Assert.Null(filter.PrimaryEntity);
+        Assert.Null(filter.Mode);
+        Assert.Null(filter.HasException);
+        Assert.Null(filter.MinDurationMs);
+        Assert.Null(filter.MaxDurationMs);
+        Assert.Null(filter.OperationType);
+        Assert.Null(filter.CorrelationId);
+        Assert.Null(filter.OrderBy);
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Cli.Tests/Tui/Dialogs/PluginTraceFilterTests.cs:23]: Fixed in c2132935aa22fe39ff5b779ea61890989ee5fdf7 — Added Assert.Null for CreatedAfter and CreatedBefore in DefaultFilter_AllPropertiesNull.

File: tests/PPDS.Cli.Tests/Tui/Dialogs/ProfileCreationDialogStateTests.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+namespace PPDS.Cli.Tests.Tui.Dialogs;
+
+[Trait("Category", "TuiUnit")]
+public sealed class ProfileCreationDialogStateTests
+{
+    [Fact]
+    public void StateRecord_CapturesAllProperties()
+    {
+        var methods = new List<string> { "Device Code", "Browser", "Client Secret" };
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Cli.Tests/Tui/Dialogs/ProfileCreationDialogStateTests.cs]: Fixed in c2132935aa22fe39ff5b779ea61890989ee5fdf7 — Switched new List<string> { ... } to a C# 12 collection expression for consistency with other tests in the file.

File: tests/PPDS.Cli.Tests/Tui/Dialogs/ProfileCreationDialogStateTests.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            Title: "Create Profile", ProfileName: "p1",
+            SelectedAuthMethod: "Device Code", AvailableAuthMethods: methods,
+            IsCreating: false, ValidationError: null, CanCreate: true);
+
+        var b = new ProfileCreationDialogState(
+            Title: "Create Profile", ProfileName: "p1",
+            SelectedAuthMethod: "Device Code", AvailableAuthMethods: methods,
+            IsCreating: false, ValidationError: null, CanCreate: true);
+
+        Assert.Equal(a, b);
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Cli.Tests/Tui/Dialogs/ProfileCreationDialogStateTests.cs:80]: Not applicable — Test passes today because both records receive the same IReadOnlyList<string> reference, which is the documented behavior of compiler-generated record equality on List<T>. Switching to distinct list instances would require changing AvailableAuthMethods on the production record from IReadOnlyList<string> to ImmutableArray<string> (or overriding Equals) â€” a production-code refactor outside the scope of this test-only PR. The current assertion still validates that records produce equal/hash-equal results for identical inputs.

File: tests/PPDS.Cli.Tests/Tui/Screens/PluginTreeNodeTests.cs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        var child = new PluginTreeNode
+        {
+            NodeType = "assembly",
+            DisplayName = "ChildAsm"
+        };
+
+        parent.Children.Add(child);
+
+        Assert.Single(parent.Children);
+        Assert.Equal("ChildAsm", ((PluginTreeNode)parent.Children[0]).DisplayName);
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply tests/PPDS.Cli.Tests/Tui/Screens/PluginTreeNodeTests.cs:153]: Not applicable — PluginTreeNodeTests.cs does not use FluentAssertions â€” every assertion is xUnit Assert.* and there is no `using FluentAssertions;` import. The Should().BeOfType<T>().Subject pattern is irrelevant here; the (PluginTreeNode) cast is appropriate xUnit style for downcasting an ITreeNode. Suggestion is based on a false premise about the file's testing style.
